File: code/zato-web-admin/src/zato/admin/static/js/channel/stomp.js

// /////////////////////////////////////////////////////////////////////////////

$.fn.zato.data_table.ChannelSTOMP = new Class({
    toString: function() {
        var s = '<ChannelSTOMP id:{0} name:{1} is_active:{2}';
        return String.format(s, this.id ? this.id : '(none)',
                                this.name ? this.name : '(none)',
                                this.is_active ? this.is_active : '(none)');
    }
});

// /////////////////////////////////////////////////////////////////////////////

$(document).ready(function() {
    $('#data-table').tablesorter();
    $.fn.zato.data_table.class_ = $.fn.zato.data_table.ChannelSTOMP;
    $.fn.zato.data_table.new_row_func = $.fn.zato.channel.stomp.data_table.new_row;
    $.fn.zato.data_table.parse();
    $.fn.zato.data_table.setup_forms(['name', 'address', 'proto_version', 'timeout', 'service_name', 'sub_to']);
})

$.fn.zato.channel.stomp.create = function() {
    $.fn.zato.data_table._create_edit('create', 'Create a new STOMP channel', null);
}

$.fn.zato.channel.stomp.edit = function(id) {
    $.fn.zato.data_table._create_edit('edit', 'Update the STOMP channel', id);
}

$.fn.zato.channel.stomp.data_table.new_row = function(item, data, include_tr) {
    var row = '';

    if(include_tr) {
        row += String.format("<tr id='tr_{0}' class='updated'>", item.id);
    }

    var is_active = item.is_active == true;

    var username = item.username;
    if(!username) {
        username = "<span class='form_hint'>(None)</span>";
    }

    row += "<td class='numbering'>&nbsp;</td>";
    row += "<td class='impexp'><input type='checkbox' /></td>";
    row += String.format('<td>{0}</td>', item.name);
    row += String.format('<td>{0}</td>', is_active ? 'Yes' : 'No');
    row += String.format('<td>{0}</td>', item.address);
    row += String.format('<td>{0}</td>', username);
    row += String.format('<td>{0}</td>', item.proto_version);
    row += String.format('<td>{0}</td>', String.format("<a href='javascript:$.fn.zato.data_table.change_password({0})'>Change password</a>", item.id));
    row += String.format('<td>{0}</td>', String.format("<a href=\"javascript:$.fn.zato.channel.stomp.edit('{0}')\">Edit</a>", item.id));
    row += String.format('<td>{0}</td>', String.format("<a href='javascript:$.fn.zato.channel.stomp.delete_({0});'>Delete</a>", item.id));
    row += String.format('<td>{0}</td>', String.format("<a href=\"javascript:$.fn.zato.data_table.ping('{0}')\">Ping</a>", item.id));
    row += String.format("<td class='ignore item_id_{0}'>{0}</td>", item.id);
    row += String.format("<td class='ignore'>{0}</td>", is_active);
    row += String.format("<td class='ignore'>{0}</td>", item.timeout);
    row += String.format("<td class='ignore'>{0}</td>", item.username ? item.username: '');

    if(include_tr) {
        row += '</tr>';
    }

    return row;
}

$.fn.zato.channel.stomp.delete_ = function(id) {
    $.fn.zato.data_table.delete_(id, 'td.item_id_',
        'STOMP channel [{0}] deleted',
        'Are you sure you want to delete the STOMP channel [{0}]?',
        true);
}

File: code/zato-web-admin/src/zato/admin/web/views/channel/stomp.py
# -*- coding: utf-8 -*-

"""
Copyright (C) 2019, Zato Source s.r.o. https://zato.io

Licensed under LGPLv3, see LICENSE.txt for terms and conditions.
"""

from __future__ import absolute_import, division, print_function, unicode_literals

# stdlib
import logging

# Django
from django.http import HttpResponse, HttpResponseServerError

# Zato
from zato.admin.web.forms import ChangePasswordForm
from zato.admin.web.forms.channel.stomp import CreateForm, EditForm
from zato.admin.web.views import change_password as _change_password, CreateEdit, Delete as _Delete, id_only_service, \
     Index as _Index, method_allowed
from zato.common.odb.model import ChannelSTOMP

logger = logging.getLogger(__name__)

class Index(_Index):
    method_allowed = 'GET'
    url_name = 'channel-stomp'
    template = 'zato/channel/stomp.html'
    service_name = 'zato.channel.stomp.get-list'
    output_class = ChannelSTOMP
    paginate = True

    class SimpleIO(_Index.SimpleIO):
        input_required = ('cluster_id',)
        output_required = ('id', 'name', 'is_active', 'address', 'username',
            'proto_version', 'timeout', 'sub_to', 'service_id', 'service_name')
        output_repeated = True

    def handle(self):
        return {
            'change_password_form': ChangePasswordForm(),
            'create_form': CreateForm(),
            'edit_form': EditForm(prefix='edit'),
        }

class _CreateEdit(CreateEdit):
    method_allowed = 'POST'

    class SimpleIO(CreateEdit.SimpleIO):
        input_required = ('name', 'is_active', 'address', 'username',
            'proto_version', 'timeout', 'sub_to', 'service_id', 'service_name')
        output_required = ('id',)

    def success_message(self, item):
        return 'Successfully {0} the channel STOMP connection [{1}]'.format(self.verb, item.name)

class Create(_CreateEdit):
    url_name = 'channel-stomp-create'
    service_name = 'zato.channel.stomp.create'

class Edit(_CreateEdit):
    url_name = 'channel-stomp-edit'
    form_prefix = 'edit-'
    service_name = 'zato.channel.stomp.edit'

class Delete(_Delete):
    url_name = 'channel-stomp-delete'
    error_message = 'Could not delete the channel STOMP connection'
    service_name = 'zato.channel.stomp.delete'

@method_allowed('POST')
def change_password(req):
    return _change_password(req, 'zato.channel.stomp.change-password')

@method_allowed('POST')
def ping(req, id, cluster_id):
    ret = id_only_service(req, 'zato.channel.stomp.ping', id, 'Could not ping the STOMP connection, e:`{}`')
    if isinstance(ret, HttpResponseServerError):
        return ret
    return HttpResponse(ret.data.info)
